packages/cw-schema-codegen/src/typescript/mod.rs

```diff
@@ -0,0 +1,160 @@
+use self::template::{
+    EnumTemplate, EnumVariantTemplate, FieldTemplate, StructTemplate, TypeTemplate,
+};
+use std::{borrow::Cow, io};
+
+pub mod template;
+
+fn expand_node_name<'a>(
+    schema: &'a cw_schema::SchemaV1,
+    node: &'a cw_schema::Node,
+) -> Cow<'a, str> {
+    match node.value {
+        cw_schema::NodeType::Array { items } => {
+            let items = &schema.definitions[items];
+            format!("z.array({})", expand_node_name(schema, items)).into()
+        }
+        cw_schema::NodeType::Float => "z.number()".into(),
+        cw_schema::NodeType::Double => "z.number()".into(),
+        cw_schema::NodeType::Boolean => "z.boolean()".into(),
+        cw_schema::NodeType::String => "z.string()".into(),
+        cw_schema::NodeType::Integer { .. } => "z.string().or(z.number())".into(), // TODO: is this correct?
```

Comment:
It seems with the way this whole thing is set up, we cannot distinguish between stringified integers (like `Uint128`, etc.) and normal ones (like `u128`). Probably should have caught that during the review without codegen.